### PR TITLE
Filter routes by geo coordinate of source/destination region

### PR DESCRIPTION
**Pull request by nilyibo**
Because geo coordinates from IP-to-geo dataset can be sometimes inaccurate, we can filter out the incorrect routes if the geolocation of first hop or last hop deviate too much from the ground truth geolocation (of source or destination region, respectively).

Fortunately the geo coordinate of each region is a single point across the entire dataset (calculated using a script previously committed, using the same IP-to-geo dataset), so we can easily filter it by comparing the first and last hop of a route with the corresponding ground truth geolocation.

### CODE REVIEW

File: analysis/itdk_geo.py [hunk 1]
```diff
@@ -38,8 +42,10 @@ def get_node_ids_with_geo_coordinates():
     node_geo_df = parse_node_geo_as_dataframe()
     return node_geo_df.index.tolist()
 
-def convert_routes_from_ip_to_latlon(routes, node_ip_to_id, node_geo_df, output_file: Optional[str]):
-    logging.info('Converting routes from IPs to lat/lons ...')
+def convert_routes_from_ip_to_latlon(routes, node_ip_to_id, node_geo_df,
```
Comment (actomlinson) [inline analysis/itdk_geo.py]: Typing also has List and Tuple.

**Review by actomlinson — APPROVED** [review]
Looks good!

File: analysis/itdk_geo.py (hunk 1 above)
Comment (actomlinson) [reply analysis/itdk_geo.py]: In general I'm supportive of typing all code but I will leave it up to you!

Comment (nilyibo) [reply analysis/itdk_geo.py]: Agree. I think this one might be one of the older code that I didn't add type checking. I'll update it.

**Pull request merged**